lib/clusterManager.js

```diff
@@ -189,7 +195,82 @@ if (cluster.isMaster) {
     this.overrideFileConfig = override;
   }
 
-   ClusterManager.prototype.notifyWorkersForAddingPlugin = function(resultHandler, indexInCluster) {
+  /* Having gotten here, we are already at the cluster, but it's safer to validate anyway
+    i. e. "if (this.storage) ..." since only the master cluster has storage */
+  ClusterManager.prototype.getStorageByKey = function(pluginId, key, resultHandler) {
+    if (this.storage) {
+      return resultHandler(this.storage.getStorageByKey(key, pluginId));
+    } else {
+      return resultHandler(null);
+    }
+  }
+
+  ClusterManager.prototype.getStorageAll = function(id, resultHandler) {
+    if (this.storage) {
+      return resultHandler(this.storage.getStorageAll(id));
+    } else {
+      return resultHandler(zluxUtil.DataserviceStorage(id).getStorageAll());
+    }
+  }
+
+  ClusterManager.prototype.getStorageCluster = function(resultHandler) {
+    if (this.storage) {
```

Comment:
All of these if-elses  should not happen. If you do not have this.storage at this time, you should stop the server with a message saying what went wrong (storage didnt exist) and how to fix it (increase timeout in server config file)